src/components/translation-builder/translation-builder.component.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import {
+  InlineLoading,
+  InlineNotification,
+  IconButton,
+  Tabs,
+  Tab,
+  TabList,
+  Dropdown,
+  ComposedModal,
+} from '@carbon/react';
+import { Download, Edit, ArrowRight } from '@carbon/react/icons';
+import { useLanguageOptions } from '@hooks/getLanguageOptionsFromSession';
+import EditTranslationModal from './edit-translation.modal';
+import { extractTranslatableStrings } from '../../utils/extractTranslatableStrings';
+
+import styles from './translation-builder.module.scss';
+interface TranslationBuilderProps {
+  formSchema: any;
+  onUpdateSchema: (updatedSchema: any) => void;
+}
+
+const TranslationBuilder: React.FC<TranslationBuilderProps> = ({ formSchema, onUpdateSchema }) => {
+  const { t } = useTranslation();
+  const { formId } = useParams<{ formId: string }>();
+  const languageOptions = useLanguageOptions();
+  const [selectedLanguageCode, setSelectedLanguageCode] = useState(() => languageOptions[0]?.code ?? 'en');
+  const [translations, setTranslations] = useState<Record<string, string>>({});
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [editingKey, setEditingKey] = useState<string | null>(null);
+  const [isModalOpen, setModalOpen] = useState(false);
+
+  const langCode = selectedLanguageCode;
+
+  useEffect(() => {
+    if (!formSchema) return;
+
+    const schemaTranslations = formSchema.translations?.[langCode];
+
+    if (schemaTranslations) {
+      setTranslations(schemaTranslations);
+    } else {
+      const fallbackStrings = extractTranslatableStrings(formSchema);
+      setTranslations(fallbackStrings);
+    }
+  }, [formSchema, langCode]);
+
+  const handleUpdateValue = (key: string, newValue: string) => {
+    const updatedTranslations = { ...translations, [key]: newValue };
+    setTranslations(updatedTranslations);
+
+    if (formSchema) {
+      const updatedSchema = { ...formSchema };
+      if (!updatedSchema.translations) {
+        updatedSchema.translations = {};
+      }
+      updatedSchema.translations[langCode] = updatedTranslations;
+      onUpdateSchema(updatedSchema);
+    }
+  };
+
+  const openEditModal = (key: string) => {
+    setEditingKey(key);
+    setModalOpen(true);
+  };
+
+  const closeEditModal = () => {
+    setEditingKey(null);
+    setModalOpen(false);
+  };
```

Comment:
Modals should be triggered using the modal system and not using state, please refer these docs on how to do so - https://o3-docs.openmrs.org/docs/modal-system